### PR TITLE
Scholarsphere upload workflow: Version checker, permissions alerts, and autofill

### LINKED ISSUE #111: Show depositor default embargo terms and other sharing rules during upload process
During upload process, query the OA Button Permissions API (or Sherpa Romeo) and display default sharing rules for the particular article version. Sharing rules can include an embargo, a license, and/or a set statement that should accompany deposit. Allow depositor to override these defaults. For items not found in the OAB Permissions API, encourage depositor to contact openaccess@psu.edu.

Implementing shareyourpaper.org (#73) would probably take care of this. Update: We decided not to do that.

### LINKED ISSUE #467: Assess article version during deposit process
When a user uploads a file to deposit, attempt to check the article version. If one can be determined, display it to the user and give deposit advice accordingly (this could be very general or could be based on #111 ). If version cannot be determined, tell the user that and perhaps allow them to indicate an article version for the purposes of #111. If user is trying to upload something other than an accepted manuscript or a final published version, encourage them not to. (OA policy requires accepted manuscript. Final published version is even better, so we'll take that if they are allowed to share it.)

### PR DESCRIPTION
**Pull request by ajkiessl**
closes #111 
closes #467

### CODE REVIEW

File: spec/component/importers/unpaywall_publication_importer_spec.rb [hunk 1]
```diff
@@ -232,7 +232,7 @@
             expect(oal.license).to be_nil
             expect(oal.oa_date).to be_nil
             expect(oal.source_updated_at).to eq '2019-12-26T19:24:13.434842'.to_datetime
-            expect(oal.version).to eq 'publishedVersion'
+            expect(oal.version).to eq I18n.t('file_versions.published_version')
```
[the remark this answers is not in the document]
Comment (ajkiessl) [reply spec/component/importers/unpaywall_publication_importer_spec.rb:235]: Oh yeah probably.  I'll change them back.

**Review by Smullz622 — APPROVED** [review]
lgtm!

**Review by ajkiessl — COMMENTED** [review]
This is all looking good to me, so I will merge this to main tomorrow morning.

**Review by banukutlu — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**